content/general/events.textile

```diff
@@ -49,6 +49,17 @@ At present, in addition to support for any custom HTTP endpoint, we have ready-m
 * "Cloudflare Workers":/general/events/cloudflare
 * "Zapier":/general/events/zapier
 
+h2(#sources). Sources
+
+Ably currently supports the following sources for all rule types:
+
+* "Messages":/realtime/messages#messages - messages trigger function calls as soon as they are published on a channel.
+* "Presence":/realtime/presence - presence events trigger function calls when clients enter, update their data, or leave channels.
+* "Channel lifecycle":/realtime/channels#channel-lifecycle – get notified when a channel is created (following the first client attaching to this channel) or discarded (when there are no more clients attached to the channel).
+* "Occupancy":/realtime/inband-occupancy#occupancy-events - occupancy events relate to the number and type of occupants in the channel.
+
+For scalability it is recommended you create rules that specify these sources, rather than creating rules that specify metachannels.
```

Comment:
For `channel.lifecycle` and `channel.occupancy` metadata in particular, for scalability reasons it is recommended that your rules use these source, rather than creating `channel.message` rules on the corresponding metachannels.